### ISSUE TITLE
Unnecessary error for "possibly undefined variable" after calling function returning `never`

### ISSUE THREAD

**Issue by DanTup** [Microsoft/TypeScript#29802]
I'm not sure if there's already an issue for this because I wasn't really sure how to describe it (hence the poor title). Given this code:

```ts
const someVar: number | undefined = undefined;

if (!someVar) {
	process.exit(0);
}

print(someVar.toString());
```

TypeScript (3.3.1) reports that `someVar` may be undefined on the last line. However it knows that `process.exit` returns `never` so it seems like it could know better. If I add ` throw` on the line after `process.exit` then the error goes away, so I guess there is some capability to know the code can never execute that applies to `throw` but not when calling functions that return `never`.

**Comment by RyanCavanaugh** [comment]
Duplicate #8655

**Comment by DanTup** [comment]
That issue is closed as design-limitation - does that mean this isn't fixable either? (I'm not complaining, just want to know if I should turn my temporary workaround into something nicer :-))
